src/Hosting/Server/ServerList.php

```diff
@@ -109,4 +109,86 @@ public function getLowestNumberedServer()
 
         return $lowestServer;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDatabaseServers()
+    {
+        $databaseServers = new DatabaseServerList();
+
+        /** @var ServerInterface $server */
+        foreach ($this as $server) {
+            if ($server->isDatabaseServer()) {
+                $databaseServers->append($server);
+            }
+        }
+
+        return $databaseServers;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Cloud API doesn't explicitly tell if the server is a fileserver, so
+     * make an educated guess based on the server name or other services that
+     * are provided.
+     */
+    public function getFileServers()
+    {
+        $fileServers = new FileServerList();
+        $databaseServers = new FileServerList();
```

Comment:
Even though we're collecting dbs, we intend to use the result as FileServerList.